src/Avalonia.Base/Input/AccessKeyHandler.cs

```diff
@@ -3,27 +3,63 @@
 using System.Linq;
 using Avalonia.Interactivity;
 using Avalonia.LogicalTree;
+using Avalonia.Threading;
+using Avalonia.VisualTree;
 
 namespace Avalonia.Input
 {
-    /// <summary>
-    /// Handles access keys for a window.
-    /// </summary>
+    // /// <summary>
+    // /// Handles access keys for a window.
+    // /// </summary>
     internal class AccessKeyHandler : IAccessKeyHandler
     {
+        private enum ProcessKeyResult
+        {
+            NoMatch,
+            MoreMatches,
+            LastMatch
+        }
+
+        private struct AccessKeyInformation
```

Comment:
In the WPF this held not only the target but also also the scope. I thought I keep it that way in case we want to attached additional information in the future. 
Futhermore, `GetInfoForElement` was renamed to `GetTargetForElement`.